Mathlib/CategoryTheory/MorphismProperty/Limits.lean

```diff
@@ -234,11 +248,20 @@ lemma IsStableUnderLimitsOfShape.lim_map {J : Type*} [Category J]
     W (lim.map f) :=
   hW X Y _ _ (limit.isLimit X) (limit.isLimit Y) f hf
 
+lemma IsStableUnderColimitsOfShape.colim_map {J : Type*} [Category J]
+    (hW : W.IsStableUnderColimitsOfShape J) {X Y : J ⥤ C}
+    (f : X ⟶ Y) [HasColimitsOfShape J C] (hf : W.functorCategory _ f) :
+    W (colim.map f) :=
+  hW X Y _ _ (colimit.isColimit X) (colimit.isColimit Y) f hf
+
 variable (W)
 
 /-- The property that a morphism property `W` is stable under products indexed by a type `J`. -/
 abbrev IsStableUnderProductsOfShape (J : Type*) := W.IsStableUnderLimitsOfShape (Discrete J)
 
+/-- The property that a morphism property `W` is stable under coproducts indexed by a type `J`. -/
+abbrev IsStableUnderCoproductsOfShape (J : Type*) := W.IsStableUnderColimitsOfShape (Discrete J)
+
 lemma IsStableUnderProductsOfShape.mk (J : Type*)
```

Comment:
We should use the following better phrasing for this lemma and its dual, which can be used in the proof of `llp_isStableUnderCoproductsOfShape`:
```lean
lemma IsStableUnderProductsOfShape.mk (J : Type*) [W.RespectsIso]
    (hW : ∀ (X₁ X₂ : J → C) [HasProduct X₁] [HasProduct X₂]
      (f : ∀ j, X₁ j ⟶ X₂ j) (_ : ∀ (j : J), W (f j)),
      W (Limits.Pi.map f)) : W.IsStableUnderProductsOfShape J := by
  sorry

lemma IsStableUnderProductsOfShape.mk (J : Type*)
    [W.RespectsIso]
    (hW : ∀ (X₁ X₂ : J → C) [HasProduct X₁] [HasProduct X₂]
      (f : ∀ j, X₁ j ⟶ X₂ j) (_ : ∀ (j : J), W (f j)),
      W (Limits.Pi.map f)) : W.IsStableUnderProductsOfShape J := by
  sorry
```